### PR TITLE
[Android] Fix issue invoking ReturnCommand

### LINKED ISSUE #13921: [regression/7.0.0] Entry Return command not firing up - .NET MAUI MVVM on Android
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Entry-Return-command-not-firing-up---NE/10289718)._

---
[severity:It's more difficult to complete my work] [regression] [worked-in:17.4.4]
Hello,
I have another issue - .NET MAUI (using NET 7.0) Entry stops firing up Return command on Android 11 device when HW return key is pressed - device have HW keyboard. Works with Android SW keyboard, but using HW keyboard on device or return key on Vysor remote control doesn't work. 

Sample project files attached.

---
#### Feedback Bot on 2/23/2023, 05:49 PM: 

(private comment, text removed)
#### bohdan on 3/1/2023, 05:05 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

### Reproduction
Repro is attached to internal VS ticket

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix issue invoking **ReturnCommand** on Android Entry.

fix-13921

To test the changes, launch the .NET MAUI Gallery on Android and navigate to the Entry samples. Select the ReturnCommand Entry, write something and tap Enter using a physical keyboard (not use the virtual).

### Issues Fixed

Fixes #13921

### CODE REVIEW

**Comment by Simoneye1011** [comment]
did you use xamarin or maui?

**Comment by jsuarezruiz** [comment]
> did you use xamarin or maui?

What do you mean exactly?. In this case, it's a PR for and using .NET MAUI.

**Comment by mattleibow** [comment]
@tj-devel709 @PureWeen @rachelkang this does keyboard things... Not sure if there are rules on this?

File: src/Core/src/Handlers/Entry/EntryHandler.Android.cs [hunk 1]
```diff
@@ -224,5 +216,54 @@ internal void HideClearButton()
...
+
+		void EditorAction(ImeAction actionId, KeyEvent? e)
+		{
+			var returnType = VirtualView?.ReturnType;
+
+			if (returnType != null)
+			{
+				var currentInputImeFlag = returnType.Value.ToPlatform();
+
+				if (actionId.IsCompletedAction(e, currentInputImeFlag))
```
Comment (jsuarezruiz) [inline src/Core/src/Handlers/Entry/EntryHandler.Android.cs:228]: https://github.com/xamarin/Xamarin.Forms/blob/9e230ad1f31b7cee2003035ecf5c4bc1b27a4c0e/Xamarin.Forms.Platform.Android/Renderers/EntryRenderer.cs#L95

**Review by rachelkang — COMMENTED** [review]
Investigating... I'm seeing the ReturnCommand gets invoked fine without any of the XF focus manipulation code, which I don't love. For one thing, the focus behavior is different when we tap the Done button vs. click the Enter key; I'm thinking whatever behavior we decide on, it should be the same (even if that's to do nothing).

Trying to see what Android does, but having some issues getting to work at the moment. If anyone has better luck, please share

**Comment by jsuarezruiz** [comment]
> Investigating... I'm seeing the ReturnCommand gets invoked fine without any of the XF focus manipulation code, which I don't love. For one thing, the focus behavior is different when we tap the Done button vs. click the Enter key; I'm thinking whatever behavior we decide on, it should be the same (even if that's to do nothing).
> 
> Trying to see what Android does, but having some issues getting to work at the moment. If anyone has better luck, please share

I have removed the specific code related with the focus management (from Xamarin.Forms):

```
View? nextFocus = null;

if (currentInputImeFlag == ImeAction.Next)
{
	nextFocus = PlatformView.FocusSearch(FocusSearchDirection.Forward);
}

if (nextFocus != null)
{
	nextFocus.RequestFocus();

	if (!nextFocus.OnCheckIsTextEditor())
	{
		PlatformView.HideKeyboard();
	}
}
else
{
	PlatformView.ClearFocus();
	PlatformView.HideKeyboard();
}
```

But, leave the changes to allow to invoke the **ReturnCommand** correctly.

**Pull request closed without merging**